program.py

```diff
@@ -278,13 +280,60 @@ def main(test=False):
             first_time_tank_2, last_time_tank_2)
         print(results)
 
+        #######################################################################
+        # Community steps
+        #######################################################################
+        if (community_id):
+            # step 12
+            print()
+            print('Get tenant roles')
+            roles = sds_client.Roles.getRoles()
+            role: Role = None
+            for r in roles:
+                if r.RoleTypeId == sds_client.Roles.CommunityMemberRoleTypeId and r.CommunityId == community_id:
+                    role = r
+                    break
+            print('Community member Id:')
+            print(role.Id)
```

Comment:
It looks like if a user were to add a community ID but the client doesnt have community member permission, role will stay as none, and the sample will fail when it tries to pull the .Id property of a null object. I don't really have a problem with that since it's a sample and it says in the readme to make sure the client has the role, but it might be more helpful to check for null at this point and output a more helpful error message saying something like "community member role not found for this client" before exiting.

Reply:
So it actually wouldn't fail at this step, it would fail at `sds_client.Streams.patchStreamAccessControl`. Even if the client doesn't have permission, the client can still see the whole list of roles which is what it's asking for with `sds_client.Roles.getRoles()`. It isn't blocked from Read access to the community member role itself, but it'd be blocked when it tries to share the stream. (The client also has to be Community Administrator, not just Community Member)

Reply:
I mean that if it doesn't find any roles with community member role type id, 'role' will still be null since no assignment will occur in the for loop. then the print statement tries to access the Id property of null. 

But that's a good point about Community Admin; the readme just says Community Member, so that should probably get mentioned there as well.

Reply:
The only reason it wouldn't find any roles with the community member role type ID is if the community didn't exist, though. Even if the client has no permissions in the community, it can still see the roles associated with the community.

But yeah, I just realized I'd gotten the README wrong, I think that means I need to update all those regardless.

Reply:
To be clear - the code is not getting the list of roles granted to the client, the code is getting the whole list of roles from the tenant itself, so whether the client has access or not, the roles should be somewhere in the list it gets back.

Reply:
Followup - I'm wrong, the client actually only needs Community Member in order to share and read streams, so the README is correct.